### PR TITLE
feature: backend Flask server connected to Rasa and StackOverflow

### LINKED ISSUE #8: Basic backend setup
## Description

Set up backend, configure any settings (linter/type checker/etc.), and choose chatbot library

## Acceptance Criteria

- `backend` folder in repo containing backend code

## Definition of Ready

- No external dependencies

## Definition of Done

- [ ] PR with all changes that is ready to review and merge
- [ ] Running in pre-prod environment correctly
- [ ] Story acceptance demo synchronously/async

## Technical Requirements

- Chatbot library
- HTTP server library

### PR DESCRIPTION
**Pull request by sunnyguan**
Set up Flask backend with ability to either use StackOverflow or query Rasa server. Assumes Rasa server is hosted on the same machine on port 5005. StackOverflow logic is Google searching for the question filtering for only StackOverflow results, then retrieve the first one and the best answer from there. 

Deployed on VM, Flask server is running on localhost:5000, ngrok (free) is used to forward this to a public ngrok endpoint, which is then used by the front-end. Rasa server (one `rasa run` and one `rasa run actions`) is running on localhost:5005, only accessible by the Flask server. 

Closes #8 and #20.

Next steps:
- [ ] #26 
- [ ] Sync to server on git push

### CODE REVIEW

File: chat_server/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+importlib-metadata==4.12.0
+itsdangerous==2.1.2
+Jinja2==3.1.2
+MarkupSafe==2.1.1
+requests==2.28.1
+six==1.16.0
+soupsieve==2.3.2.post1
+urllib3==1.26.12
+Werkzeug==2.2.2
+zipp==3.8.1
```
Comment (BryantH24) [inline chat_server/requirements.txt:19]: nit: rasa-model uses env.yml

Comment (BryantH24) [reply chat_server/requirements.txt:19]: (probably not worth changing though)

**Review by BryantH24 — APPROVED** [review]
LGTM!

File: chat_server/requirements.txt (hunk 1 above)
Comment (sunnyguan) [reply chat_server/requirements.txt:19]: Will add new issue for me to learn how to use conda

**Pull request merged**